Neo4j.Driver/Neo4j.Driver/Internal/ConcurrentSet.cs

```diff
@@ -0,0 +1,49 @@
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace Neo4j.Driver.Internal
+{
+    internal class ConcurrentSet <T> : IEnumerable<T>
```

Comment:
We could, then we need to impl 20+ methods on ISet interface. I feel it is not needed.